workers_launch_template.tf

```
@@ -141,7 +141,13 @@ resource "aws_autoscaling_group" "workers_launch_template" {
           version = lookup(
             var.worker_groups_launch_template[count.index],
             "launch_template_version",
-            local.workers_group_defaults["launch_template_version"],
+            lookup(
+              var.worker_groups_launch_template[count.index],
+              "use_latest_version",
+              local.workers_group_defaults["use_latest_version"]
+            )
+            ? aws_launch_template.workers_launch_template.*.latest_version[count.index]
+            : aws_launch_template.workers_launch_template.*.default_version[count.index]
```

```suggestion
            lookup(
              var.worker_groups_launch_template[count.index],
              "launch_template_version",
              local.workers_group_defaults["launch_template_version"]
            ) == "$Latest"
            ? aws_launch_template.workers_launch_template.*.latest_version[count.index]
            : aws_launch_template.workers_launch_template.*.default_version[count.index]
```

We always create LT for self-managed worker groups, so there are no reason to allow external version. This was first implemented for Managed Node Group, to let users provide external LT for MNG (only).